docs/src/main/asciidoc/using-vertx.adoc

```diff
@@ -0,0 +1,483 @@
+
+Navigate to http://localhost:8080/streaming.html.
+A new greeting should show-up every 2 seconds.
+
+[source, text]
+----
+Hello Quarkus! (Thu Mar 21 17:26:12 CET 2019)
+
+Hello Quarkus! (Thu Mar 21 17:26:14 CET 2019)
+
+Hello Quarkus! (Thu Mar 21 17:26:16 CET 2019)
+
+...
+----
+
+=== Using JSON results
+
+Vert.x API heavily relies on JSON, namely the `io.vertx.core.json.JsonObject` and `io.vertx.core.json.JsonArray` types.
+They are both supported as {project-name} web resource return types.
+
+Consider these endpoints:
+
+[source,java]
+----
+@Path("/hello")
+@Produces(MediaType.APPLICATION_JSON)
+public class VertxJsonResource {
+
+    @GET
+    @Path("{name}/object")
+    public JsonObject jsonObject(@PathParam("name") String name) {
+        return new JsonObject().put("Hello", name);
+    }
+
+    @GET
+    @Path("{name}/array")
+    public JsonArray jsonArray(@PathParam("name") String name) {
+        return new JsonArray().add("Hello").add(name);
+    }
+}
+----
+
+In your browser, navigate to http://localhost:8080/hello/Quarkus/object. You should see:
+
+[source, text]
+----
+{"Hello":"Quarkus"}
+----
+
+Then, navigate to http://localhost:8080/hello/Quarkus/array:
+
+[source, text]
+----
+["Hello","Quarkus"]
+----
+
+Needless to say, this works equally well when the JSON result is wrapped in a `CompletionStage` or a `Publisher`.
+
+== Using Vert.x Clients
+
+As you can inject a Vert.x instance, you can use Vert.x clients in a {project-name} application.
+This section gives an example with the `WebClient`.
+
+=== Picking the right dependency
+
+Depending on the API model you want to use you need to add the right dependency to your `pom.xml` file:
+
+[source, xml, subs=attributes+]
+----
+<!-- bare API -->
+<dependency>
+  <groupId>io.vertx</groupId>
+  <artifactId>vertx-web-client</artifactId>
+  <version>{vertx-version}</version>
+</dependency>
+
+<!-- Axle API -->
+<dependency>
+  <groupId>io.smallrye.reactive</groupId>
+  <artifactId>smallrye-axle-web-client</artifactId>
+  <version>{axle-version}</version>
+</dependency>
+
+<!-- RX Java 2 API -->
+<dependency>
+  <groupId>io.vertx</groupId>
+  <artifactId>vertx-rx-java2</artifactId>
+  <version>{vertx-version}</version>
+</dependency>
+----
+
+In this guide, we are going to use the Axle API, so:
+
+[source, xml, subs=attributes+]
+----
+<dependency>
+  <groupId>io.smallrye.reactive</groupId>
+  <artifactId>smallrye-axle-web-client</artifactId>
+  <version>{axle-version}</version>
+</dependency>
+----
+
+As we are going to invoke a `https` service, also configure the quarkus-maven-plugin from the _native_ profile with:
+
+[source, xml]
+----
+<executions>
+    <execution>
+        <goals>
+            <goal>native-image</goal>
+        </goals>
+        <configuration>
+            <enableHttpUrlHandler>true</enableHttpUrlHandler>
+            <enableHttpsUrlHandler>true</enableHttpsUrlHandler> <!-- Add this line -->
```

Comment:
Aren't those set automatically with an ssl config option?

Reply:
Good point.

Reply:
This line set native ssl to `true`, I don't know if the opposite it true, let me check.

Reply:
So, both work :-D 

I've changed to use the `application.properties`